pkgs/by-name/ma/marmite/package.nix

```
@@ -0,0 +1,48 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  oniguruma,
+  stdenv,
+  darwin,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "marmite";
+  version = "0.2.3";
+
+  src = fetchFromGitHub {
+    owner = "rochacbruno";
+    repo = "marmite";
+    rev = version;
+    hash = "sha256-AblitYe7YDUc2Tg2P7j+wnOjMAhDtieDsbq6B6x+uMs=";
+  };
+
+  cargoHash = "sha256-u8sJS9hvIao+af7IA4vE0eUSx0xmI1Kb5NXyZBrOCIw=";
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs =
+    [
+      oniguruma
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.CoreFoundation
+      darwin.apple_sdk.frameworks.CoreServices
+    ];
+
+  env = {
+    RUSTONIG_SYSTEM_LIBONIG = true;
+  };
+
+  meta = {
+    description = "A Static Site Generator for Blogs";
+    homepage = "https://github.com/rochacbruno/marmite";
+    license = lib.licenses.agpl3Only;
```

Also, I think it's agpl3Plus. (See https://github.com/rochacbruno/marmite/blob/main/Cargo.toml#L5)
```suggestion
    license = lib.licenses.agpl3Plus;
```